### PR TITLE
Add option to print LLVM IR (without debugging)

### PR DESCRIPTION
**Pull request by axic**

### CODE REVIEW

File: src/backend/llvm_ast.c [hunk 1]
```diff
@@ -432,6 +432,14 @@ struct LLVMOpaqueModule *blvm_create_module(
     if (compiler_args_print_backend_debug(ctx->args)) {
         bllvm_mod_debug(ctx->llvm_mod, mod_name);
     }
+
+    size_t mod_id_len;
+    char *mod_id = LLVMGetModuleIdentifier(ctx->llvm_mod, &mod_id_len);
```
Comment (LefterisJP) [inline src/backend/llvm_ast.c]: This seems to be available only on the latest LLVM version (3.9). I do have it locally for example, but Travis tests use use LLVM version 3.7. I would like to be able to compile Refu with 3.7 too so perhaps use the module name that is already known and exists in this same function?

It should be the cstring `mod_name` found here: https://github.com/refu-lang/refu/pull/40/files#diff-10b12b4c46f372efd867280d766ffeadL375

If you do so then you should also move `RFS_POP()` to the end of the function, both in the happy case and the fail case.

**Review by LefterisJP — CHANGES_REQUESTED** [review]

File: src/backend/llvm_ast.c (hunk 1 above)
Comment (axic) [reply src/backend/llvm_ast.c]: On my LLVM, `mod_name` always equals to `function_end` for some reason. Any ideas?

Comment (LefterisJP) [reply src/backend/llvm_ast.c]: @axic yes I know exactly why.

**TL;DR:**
So ... without going into too many details you will have to move `RFS_POP();` from line `382` to both the happy end and the unhappy end of the function. So move it to line 442 and to line 446.

**More details**: 
`RFstring`, the string we are using from the `rfbase` library is not null terminated. It was designed like that. `LLVM` needs null terminated strings. `rf_string_cstr_from_buff_or_die()` is one of the many functions that create temporary strings by utilizing a temporary buffer that is initialized in the beginning of the program. 

It saves the hustle of allocating and freeing strings but has the drawback that you have to remember where you initialized it with an `RFS_PUSH();` and when you want to stop using it use an `RFS_POP();`

In the current code the `RFS_POP();` is right after the LLVM module initialization call so after that point any other usage of the RFS buffer would overwrite it. Which happens somewhere down the line with the `function_end` label.

**Comment by axic** [comment]
Rebased against #41.

File: src/backend/llvm_ast.c (hunk 1 above)
Comment (axic) [reply src/backend/llvm_ast.c]: Fixed that bug independently in #41 and fixed this not duplicate the work.

**Review by LefterisJP — APPROVED** [review]

**Comment by LefterisJP** [comment]
Thank you @axic. Much appreciated.
